### PR TITLE
refactor: converting all _.each to native forEach (#302)

### PR DESCRIPTION
**Pull request by mexitek**
*Issue #, if available:* 302

*Description of changes:* Refactoring all uses for lodash `_.each(...)` to ES6 `[].forEach(...)`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: __tests__/utils/flattenProperties.test.js [hunk 1]
```diff
@@ -43,9 +44,9 @@ describe('utils', () => {
         properties.white
       ];
 
-      var sortedExpectedRet = _.sortBy(expected_ret, ['value']);
+      var sortedExpectedRet = expected_ret.concat().sort(sortBy("value"));
```
Comment (chazzmoney) [inline __tests__/utils/flattenProperties.test.js]: Why are we doing a concat before the sort?

**Comment by chazzmoney** [comment]
@mexitek 

Hey, appreciate the work you did here - can you speak to the concat() you are adding in? Would love to merge this in but need to understand the purpose.

Thank you!

**Comment by mexitek** [comment]
@chazzmoney [`Array.sort()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) will sort in place and not create a copy of the original array. By adding `contcat()`, we create a copy of the original array then sort that array in place.

File: __tests__/utils/flattenProperties.test.js (hunk 1 above)
Comment (mexitek) [reply __tests__/utils/flattenProperties.test.js]: Sorry for the delay. Answered below.

File: lib/common/templates/ios-swift/class.swift.template [hunk 2]
```diff
@@ -29,7 +29,7 @@
   // Filter to only those props wanted based on the filter, then sort
   // them by category so we keep like props together, then by name
   // so they are easier to find alphabetically.
-  var props = _.sortBy(allProperties, item => item.attributes.category + item.name);
+  var props = allProperties.concat().sort(item => item.attributes.category + item.name);
```
Comment (chazzmoney) [inline lib/common/templates/ios-swift/class.swift.template]: Can we use one of these two methods to create the array copy?

[...allProperties]
allProperties.slice(0)

File: lib/common/templates/ios-swift/enum.swift.template [hunk 3]
```diff
@@ -29,7 +29,7 @@
   // Filter to only those props wanted based on the filter, then sort
   // them by category so we keep like props together, then by name
   // so they are easier to find alphabetically.
-  var props = _.sortBy(allProperties, item => item.attributes.category + item.name);
+  var props = allProperties.concat().sort(item => item.attributes.category + item.name);
```
Comment (chazzmoney) [inline lib/common/templates/ios-swift/enum.swift.template]: Can we use one of these two methods to create the array copy?

[...allProperties]
allProperties.slice(0)

**Comment by chazzmoney** [comment]
@mexitek Are you still working on this? Would love to merge this in.

**Comment by dbanksdesign** [comment]
Lots of merge conflicts, probably better to start a fresh PR. Closing this for now

**Pull request closed without merging**